### PR TITLE
DOI Documentation

### PR DESCRIPTION
**Pull request by coverbeck**
**Description**
My ticket is to document to Zenodo-GitHub integration, but I found it hard to only do that on its own, so I created an overview page of DOIs and put in a paragraph about the Zenodo-GitHub integration.

Steve has SEAB-6738 to document the auto DOI generation this sprint. I figure he can either add his documentation to this page, or create a new page, and link to it from here.

**Issue**
SEAB-6757

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If the PR has any new pages, only **AFTER** the PR is approved by all reviewers, generate a new discourse topic, by running the action at https://github.com/dockstore/dockstore-documentation/actions/workflows/add-discourse-topic.yml. Otherwise we may end up with orphaned topics, which would need manual management.

### CODE REVIEW

File: docs/advanced-topics/dockstore-cli/dockstore-cli-faq.rst
Comment (coverbeck) [inline docs/advanced-topics/dockstore-cli/dockstore-cli-faq.rst:44]: Updated because linkcheck is failing. It's been failing in prod as well for a few weeks; I've been getting email notifications but not taking any action on them.

**Comment by coverbeck** [comment]
A test is failing because topic needs to be added. Waiting for approval first, per PR instructions.

**Review by denis-yuen — APPROVED** [review]

File: docs/end-user-topics/dois.rst
Comment (svonworl) [inline docs/end-user-topics/dois.rst:61]: IMHO, this bullet list [and those that follow] mostly reiterate what's already been explained, or include judgements about the utility of the feature, I'm not sure we need them.

File: docs/end-user-topics/dois.rst
Comment (svonworl) [reply docs/end-user-topics/dois.rst:10]: > I recently learned that concept DOI seems to be a zenodo concept. So far, this sentence seems to be ok even with harvested DOIs since GitHub's integration also uses zenodo, but it could be different someday if we work with other DOIs.

Yes, this is important to understand, so please permit me to reiterate:

A DOI itself, per the standard and infrastructure, is simply an identifier which is associated with some summary metadata which typically includes a link to a resource.  The DOI standard supports updating the DOI's metadata after it is created.

Zenodo has created the notion of a concept DOI and version DOI, and also archives the resource, and Zenodo-issued DOIs operate within this "semi-permanent archive" conceptual framework.

However, many [non-Zenodo] DOIs simply link directly to the original resource (paper PDF on the publishers site, for example), and there is no archival copy made.

Comment (denis-yuen) [reply docs/end-user-topics/dois.rst:10]: To be clear, I think concept DOIs (the concept of organizing and linking versioned DOIs) are important and worth bringing up in this document (as are those archives). 

I'm just bringing this up in the context of making the doc future-proof.

Comment (coverbeck) [reply docs/end-user-topics/dois.rst:10]: Hmm, I did not know this. I will tailor the doc accordingly.

File: docs/end-user-topics/dois.rst
Comment (coverbeck) [reply docs/end-user-topics/dois.rst:45]: Per that link, I would have been in the majority until 2008!

Comment (svonworl) [reply docs/end-user-topics/dois.rst:45]: Was always taught that compound words started out separate ("trade offs"), then, as language evolved, became hyphenated ("trade-offs"), and then finally joined ("tradeoffs"), never thought about it happening the other way around...

File: docs/end-user-topics/dois.rst
Comment (coverbeck) [reply docs/end-user-topics/dois.rst]: Reworked

File: docs/end-user-topics/dois.rst
Comment (coverbeck) [reply docs/end-user-topics/dois.rst]: Reworded to not have `tradeoffs` at all.

File: docs/end-user-topics/dois.rst
Comment (kathy-t) [inline docs/end-user-topics/dois.rst:58]: Maybe intentional, but this bullet point is bolded when I view it locally 
image

File: docs/end-user-topics/dois.rst
Comment (kathy-t) [inline docs/end-user-topics/dois.rst:70]: Should mention that it's for published entries only

File: docs/end-user-topics/dois.rst
Comment (kathy-t) [inline docs/end-user-topics/dois.rst:87]: This is also bolded 
image

File: docs/end-user-topics/dois.rst
Comment (kathy-t) [inline docs/end-user-topics/dois.rst:6]: Typo: Digial -> Digital

File: docs/end-user-topics/dois.rst
Comment (kathy-t) [inline docs/end-user-topics/dois.rst:47]: Don't feel super strongly about it, but suggesting we use Overview instead of Features

File: docs/end-user-topics/dois.rst
Comment (kathy-t) [inline docs/end-user-topics/dois.rst:70]: All these bullet points are inside the note

image

File: docs/end-user-topics/dois.rst
Comment (kathy-t) [inline docs/end-user-topics/dois.rst:71]: The DOIs when users manually create it are also added to the Zenodo Dockstore community

File: docs/end-user-topics/dois.rst
Comment (kathy-t) [inline docs/end-user-topics/dois.rst:50]: Might be nice to have a brief table that summarizes when the DOIs are created, whether a user Zenodo account is needed, and if the version is snapshotted (in addition to the sections you already have). Something like

| **DOI Generation Type**    | **Zenodo Account Required** | **Snapshot Required** | **Created on**                                                                      |
|----------------------------|-----------------------------|-----------------------|-------------------------------------------------------------------------------------|
| Manually by Dockstore User | yes                         | yes                   | user request for specific versions                                                  |
| Automatically by Dockstore | no                          | no                    |  tag creation for a published entry, on publish   |
| Zenodo-GitHub Integration  | yes                         | no                    | GitHub release for a repository that is enabled for the Zenodo-GitHub integration   |

**Review by svonworl — APPROVED** [review]

File: docs/end-user-topics/dois.rst
Comment (denis-yuen) [reply docs/end-user-topics/dois.rst:50]: May be especially useful if we pursue issuing DataCite DOIs directly

File: docs/end-user-topics/dois.rst
Comment (denis-yuen) [reply docs/end-user-topics/dois.rst:73]: Think it is worth noting, there's no UI component (yet) but I think this is necessary/desirable to get consent for this feature

File: docs/end-user-topics/dois.rst
Comment (coverbeck) [reply docs/end-user-topics/dois.rst:58]: I'd noticed that earlier too further below, and thought it was a fluke... It looks like you need a blank line before the nested list item to resolve.

File: docs/end-user-topics/dois.rst
Comment (coverbeck) [reply docs/end-user-topics/dois.rst:70]: Hmm, not seeing that. Is it possible you'd made changes locally?

File: docs/end-user-topics/dois.rst
Comment (coverbeck) [reply docs/end-user-topics/dois.rst:71]: I did not know that.

File: docs/end-user-topics/dois.rst
Comment (kathy-t) [reply docs/end-user-topics/dois.rst:70]: I re-installed the requirements.txt and I don't see this anymore

File: docs/end-user-topics/dois.rst
Comment (coverbeck) [reply docs/end-user-topics/dois.rst:73]: OK, leaving as is.

**Review by kathy-t — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]

**Comment by coverbeck** [comment]
Re-requesting review in case any objections to new title. See discussion [Slack](https://ucsc-gi.slack.com/archives/C05EZH3RVNY/p1730913888563299), apparently the title was too short to generate a Discord topic.

**Review by kathy-t — APPROVED** [review]

**Pull request merged**
